testcases/OpTestInterrupt.py

```diff
@@ -0,0 +1,189 @@
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+'''
+OpTestInterrupt
+---------------
+
+This module contain testcases related to XIVE.
+
+1.Enable kernel logging, basically generate traffic
+2.Enable console traffic by printing the 
+processes/tasks to the console
+3.check Xive is configured properly or not
+4.check system reboot with xive=on
+5.check system reboot with xive=off
+6.validate interrupts handled by CPU's
+'''
+import os
+import re
+import sys
+import time
+import pexpect
+import unittest
+import subprocess
+
+import OpTestConfiguration
+from common import OpTestHMC, OpTestFSP
+from common.OpTestSOL import OpSOLMonitorThread
+from common.OpTestError import OpTestError
+from common.OpTestSystem import OpSystemState
+from common.OpTestSSH import ConsoleState as SSHConnectionState
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.Exceptions import KernelOOPS, KernelKdump
+
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class OpTestInterrupt(unittest.TestCase):
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.platform = conf.platform()
+        self.bmc_type = conf.args.bmc_type
+        self.util = self.cv_SYSTEM.util
+        self.c = self.cv_SYSTEM.console
+        self.con = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+    
+    def xive_test(self):
+        '''
+        This function will do below steps 
+        1.Enable kernel logging, basically generate traffic
+        2.Enable console traffic by printing the 
+        processes/tasks to the console
+        '''
+        self.c.run_command("echo 10 > /proc/sys/kernel/printk")
+        try:
+            i = 1
+            while i <= 10: 
+                self.c.run_command("echo t > /proc/sysrq-trigger", timeout=600)
+                i = i+1
+        except(KernelOOPS, KernelKdump):
+            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.c.run_command("echo $?")
+
+class XiveTesting(OpTestInterrupt):
+    '''
+    1.This function will check xive is configured properly or not
+    2.check for system reboot with Xive=off
+    3.check for system reboot with Xive=on
+    '''
+    def setup_test(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.con.run_command("uname -a")
+        result = self.con.run_command("grep -i --color=never xive /boot/config-`uname -r`")
+        for line in result:
+            temp = ''.join(self.con.run_command("echo %s | cut -d '=' -f 2" % line))
+            if temp != 'y':
+                self.fail("xive is not configured properly")
```

Comment:
It looks like we have multiple config names with *_XIVE.
Do you really expect all of them to be `y` ?